AdvanceDS/Assignment 1/genralize list/GenralizeList.java

```diff
@@ -0,0 +1,228 @@
+package addsAssignment;
+
+import java.util.List;
+import java.util.ArrayList;
+
+public class GenralizeList {
+	private Node start;
+	private Stack expression;
+	private int max = 0;
+	private int sum = 0;
+	private String outputList = "";
+	private boolean b = false;
+
+	public Node getStart() {
+		return start;
+
+	}
+
+	class Stack {
+		private int top;
+		private int size;
+		private Object stack[];
+
+		public Stack(int size) {
+
+			this.size = size;
+			top = -1;
+			stack = new Object[size];
+		}
+
+		public int getTop() {
+			return top;
+		}
+
+		public void setTop(int top) {
+			this.top = top;
+		}
+
+		public void push(Object item) {
+
+			if (top >= size - 1) {
+				System.out.println("Stack overflow");
+
+			} else {
+				top = top + 1;
+				stack[top] = item;
+			}
+
+		}
+
+		public Object pop() {
+			Object elementToPop = null;
+			if (size == -1) {
+				System.out.println("Stack underflow");
+			} else {
+
+				elementToPop = stack[top];
+				top--;
+
+			}
+
+			return elementToPop;
+		}
+
+		public boolean isEmpty() {
+			if (top == -1)
+				return true;
+			else
```

Comment:
If you are returning in both "if" and "else" then you can skip the "else" keyword ;)